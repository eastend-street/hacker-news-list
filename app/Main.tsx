export default function Main({ children }: { children: React.ReactNode }) {
  return <main className="flex justify-center bg-ivory p-3">{children}</main>;
}
