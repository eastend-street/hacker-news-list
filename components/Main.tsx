export default function Main({ children }: { children: React.ReactNode }) {
  return <main className="flex bg-ivory p-3">{children}</main>;
}
