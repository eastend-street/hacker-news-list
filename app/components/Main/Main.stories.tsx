import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import Main from './Main';

export default {
  title: 'components/Main',
  component: Main,
} as Meta;

const Template: StoryFn<typeof Main> = () => <Main>Main Content</Main>;

export const Default = Template.bind({});
