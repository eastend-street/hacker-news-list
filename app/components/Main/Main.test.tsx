import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as stories from './Main.stories';

const { Default } = composeStories(stories);

describe('components/Main', () => {
  test('should render Main component', () => {
    const { container } = render(<Default />);

    expect(screen.getByText(/Main Content/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
