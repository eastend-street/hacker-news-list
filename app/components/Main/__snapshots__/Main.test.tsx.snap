// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Main should render Main component 1`] = `
<div>
  <main
    class="flex bg-ivory p-3"
  >
    Main Content
  </main>
</div>
`;
